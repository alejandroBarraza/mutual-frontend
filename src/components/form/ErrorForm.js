import React from 'react';
import { Typography } from '@mui/material';

export const ErrorForm = ({ error }) => {
    return (
        <Typography variant='caption' sx={{ mt: 1, color: 'red' }}>
            {error}
        </Typography>
    );
};
