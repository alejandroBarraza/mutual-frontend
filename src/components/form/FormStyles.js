import { alpha, styled } from '@mui/material/styles';
import InputBase from '@mui/material/InputBase';
export const BootstrapInput = styled(InputBase)(({ theme }) => ({
    'label + &': {
        marginTop: theme.spacing(3),
    },
    '& .MuiInputBase-input': {
        borderRadius: 5,
        position: 'relative',
        backgroundColor: '#FAFAFA',
        border: '2px solid #ced4da',
        fontSize: '1.5rem',
        width: '100%',
        padding: '10px 12px',
        transition: theme.transitions.create(['border-color', 'background-color', 'box-shadow']),
        // Use the system font instead of the default Roboto font.
        fontFamily: [
            '-apple-system',
            'BlinkMacSystemFont',
            '"Segoe UI"',
            'Roboto',
            '"Helvetica Neue"',
            'Arial',
            'sans-serif',
            '"Apple Color Emoji"',
            '"Segoe UI Emoji"',
            '"Segoe UI Symbol"',
        ].join(','),
        '&:focus': {
            boxShadow: `${alpha(theme.palette.primary.main, 0.25)} 0 0 0 0.2rem`,
            borderColor: theme.palette.primary.main,
        },
    },
}));

export const styles = {
    outterBox: {
        minHeight: '72vh',
        width: '100%',
        position: 'relative',
        pb: '4rem',
    },
    innerBox: {
        bgcolor: '#C3D600',
        height: '36vh',
        width: '100%',
        position: 'absolute',
    },
    container: {
        position: 'relative',
        pt: '4rem',
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'column',
        alignItems: 'center',
    },
    boxTitle: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        height: '2rem',
        color: '#fff',
        fontSize: '3rem',
    },
    boxButton: {
        gridColumn: '1 / 3',
        bgcolor: '#C3D600',
        fontSize: '1.5rem',
        fontWeight: 'bold',
        color: 'white',
    },
    boxForm: {
        flexDirection: 'column',
        gridTemplateColumns: { sm: '1fr 1fr' },
        gap: 2,
        width: '90%',
        height: '100%',
        mt: '2rem',
        padding: '2rem',
        bgcolor: '#fff',
        borderRadius: '.8rem',
        boxShadow: 3,
    },
};
