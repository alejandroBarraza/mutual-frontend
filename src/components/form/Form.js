import React from 'react';

import { useForm } from 'react-hook-form';
import {
    Button,
    Container,
    Typography,
    useMediaQuery,
    InputLabel,
    FormControl,
    Box,
} from '@mui/material';
import { BootstrapInput, styles } from './FormStyles';


import { ErrorForm } from './ErrorForm';
import { API_URL } from '../utils/url';

export const Form = () => {
    const matches = useMediaQuery('(min-width:768px)');
    const {
        register,
        handleSubmit,
        formState: { errors },
    } = useForm();

    const onSubmit = (data, e) => {
        e.preventDefault();
        fetch(`${API_URL}/mensajes/customMail`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(data),
        })
            .then(() => e.target.reset())
            .catch((err) => console.log(err));
    };

    return (
        <Box sx={styles.outterBox}>
            <Box sx={styles.innerBox}></Box>
            <Container sx={styles.container}>
                <Typography component='div'>
                    <Box sx={styles.boxTitle}>Contacto</Box>
                </Typography>
                <Box
                    component='form'
                    noValidate
                    autoComplete='off'
                    onSubmit={handleSubmit(onSubmit)}
                    sx={{ display: matches ? 'grid' : 'flex', ...styles.boxForm }}>
                    <FormControl variant='standard'>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Nombre
                        </InputLabel>
                        <BootstrapInput {...register('nombre', { required: true })} />
                        {errors.nombre && <ErrorForm error='El nombre es requerido' />}
                    </FormControl>
                    <FormControl variant='standard'>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Apellido
                        </InputLabel>
                        <BootstrapInput type='text' {...register('apellido', { required: true })} />
                        {errors.apellido && <ErrorForm error='El apellido es requerido' />}
                    </FormControl>
                    <FormControl variant='standard'>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Correo
                        </InputLabel>
                        <BootstrapInput
                            type='email'
                            {...register('correo', {
                                required: {
                                    value: true,
                                    message: 'El correo es requerido',
                                },
                                pattern: {
                                    value: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
                                    message: 'Ingresar un email válido',
                                },
                            })}
                        />
                        {errors.correo && <ErrorForm error={errors.correo.message} />}
                    </FormControl>
                    <FormControl variant='standard'>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Numero de Telefono
                        </InputLabel>
                        <BootstrapInput
                            placeholder='+569 XXXX XXXX'
                            {...register('telefono', {
                                required: {
                                    value: true,
                                    message: 'el teléfono es requerido',
                                },
                                minLength: {
                                    min: 8,
                                    message: 'El número debe contener minimo 8 caracteres',
                                },
                            })}
                        />
                        {errors.telefono && <ErrorForm error={errors.telefono.message} />}
                    </FormControl>
                    <FormControl variant='standard'>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Empresa
                        </InputLabel>
                        <BootstrapInput {...register('empresa', { required: true })} />
                        {errors.empresa && <ErrorForm error='La empresa es requerida' />}
                    </FormControl>
                    <FormControl variant='standard'>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Cargo
                        </InputLabel>
                        <BootstrapInput {...register('cargo', { required: true })} />
                        {errors.cargo && <ErrorForm error='El cargo es requerido' />}
                    </FormControl>
                    <FormControl variant='standard' sx={{ gridColumn: '1 / 3' }}>
                        <InputLabel shrink htmlFor='bootstrap-input' sx={{ fontSize: '1.5rem' }}>
                            Cuentanos tus necesidades
                        </InputLabel>
                        <BootstrapInput
                            {...register('descripcion', { required: true })}
                            multiline
                            maxRows={6}
                        />
                        {errors.descripcion && <ErrorForm error='La descripción es requerida' />}
                    </FormControl>
                    <Button type='submit' variant='contained' color='success' sx={styles.boxButton}>
                        Enviar
                    </Button>
                </Box>
            </Container>
        </Box>
    );
};
