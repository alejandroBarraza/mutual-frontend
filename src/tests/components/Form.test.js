import { render } from "@testing-library/react"
import { Form } from '../../components/form/Form'
describe('test in <Form /> component', () => {

   test('renders loaded correctly', () => {
      const { container } = render(<Form />)
      expect(container).toMatchSnapshot();
   })
})