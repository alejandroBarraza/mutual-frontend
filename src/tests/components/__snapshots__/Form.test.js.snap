// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <Form /> component renders loaded correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-lw6ge8"
  >
    <div
      class="MuiBox-root css-htakcv"
    />
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-dke64x-MuiContainer-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        <div
          class="MuiBox-root css-1e98ng6"
        >
          Contacto
        </div>
      </div>
      <form
        autocomplete="off"
        class="MuiBox-root css-1ntafzr"
        novalidate=""
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Nombre
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1scu3a5-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="nombre"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Apellido
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1scu3a5-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="apellido"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Correo
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1scu3a5-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="correo"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Numero de Telefono
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1scu3a5-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="telefono"
              placeholder="+569 XXXX XXXX"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Empresa
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1scu3a5-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="empresa"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Cargo
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1scu3a5-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="cargo"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1pu5int-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1qwi02l-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="bootstrap-input"
          >
            Cuentanos tus necesidades
          </label>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-1hjnjn-MuiInputBase-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiInputBase-inputMultiline css-rhrbtd-MuiInputBase-input"
              name="descripcion"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiInputBase-inputMultiline css-rhrbtd-MuiInputBase-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
              tabindex="-1"
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-6rrm7x-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Enviar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;
